### PR TITLE
#242 AccessToken core-impl.

### PR DESCRIPTION
**Pull request by criske**
- Github and Gitlab AccessToken implemented and tested.

PR for #242

### CODE REVIEW

File: self-core-impl/src/main/java/com/selfxdsd/core/JsonResources.java [hunk 1]
```diff
@@ -214,15 +214,13 @@ private HttpRequest request(
...
+            if(this.accessToken != null) {
                 request = HttpRequest.newBuilder()
                     .uri(uri)
                     .method(method, body)
                     .header("Content-Type", "application/json")
-                    .header(
-                        "Authorization",
-                        "token " + this.accessToken
-                    ).build();
+                    .header(this.accessToken.header(), this.accessToken.value())
```
Comment (amihaiemil) [inline self-core-impl/src/main/java/com/selfxdsd/core/JsonResources.java:222]: @criske In the case of Github, you have to preppend the ``token `` word in front of the actual token's value. Maybe you can modify the ``value()`` method to return ``"token " + value``?

Comment (criske) [reply self-core-impl/src/main/java/com/selfxdsd/core/JsonResources.java:222]: @amihaiemil It's prepended in GithubAccessToken constructor.

**Comment by amihaiemil** [comment]
@rultor merge it please

**Comment by amihaiemil** [comment]
@criske there is a compilation error, apparently :D

**Comment by amihaiemil** [comment]
@rultor merge

**Pull request merged**